### PR TITLE
Add requirements for Raspberry Pi

### PR DESCRIPTION
**Pull request by darox**
Add documentation for running Cilium on a Raspberry Pi.

Signed-off-by: darox <9934402+darox@users.noreply.github.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

```release-note
docs: Add requirements for installing Cilium on Raspberry Pi
```

### CODE REVIEW

File: Documentation/operations/system_requirements.rst [hunk 1]
```diff
@@ -129,6 +129,15 @@ to ``/etc/systemd/network/01-no-dhcp.network`` and then
...
         systemctl restart systemd-networkd
 
+Raspberry Pi 
+~~~~~~~~~~~~
+
+Before running Cilium on a Raspberry Pi, please make sure to install the following package:
+
+.. code-block:: shell-session
+
+        sudo apt install linux-modules-extra-$(uname -r)
```
Comment (gentoo-root) [inline Documentation/operations/system_requirements.rst]: This looks specific to Raspberry Pi OS (ex Raspbian), while Raspberry Pi can run a variety of other Linux distributions, which may not need this command ([seems openSUSE doesn't need it](https://cilium.slack.com/archives/C53TG4J4R/p1674098237654269?thread_ts=1674036780.656589&cid=C53TG4J4R); I suppose, ArchLinuxARM doesn't need it either).

Another thought: as Raspberry Pi OS is based on Debian, and packages are built in a similar way, there is a chance this package might also be needed on other Debian-based distributions, not necessarily running on Raspberry Pi. Do you happen to know what specific kernel module was missing before installing this package?

**Review by gentoo-root — CHANGES_REQUESTED** [review]

File: Documentation/operations/system_requirements.rst [hunk 2]
```diff
@@ -129,6 +129,15 @@ to ``/etc/systemd/network/01-no-dhcp.network`` and then
         systemctl daemon-reload
         systemctl restart systemd-networkd
 
+Ubuntu 22.04 on Raspberry Pi 
+~~~~~~~~~~~~
```
Comment (qmonnet) [inline Documentation/operations/system_requirements.rst]: Netflify preview currently breaks on this:

```suggestion
Ubuntu 22.04 on Raspberry Pi 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by qmonnet — CHANGES_REQUESTED** [review]

**Comment by qmonnet** [comment]
Please, can you also consider using a real email address in the commit's `Signed-off-by:` tag? Although this doesn't happen often, we've had to reach out to some contributors in the past.

**Comment by darox** [comment]
> Please, can you also consider using a real email address in the commit's `Signed-off-by:` tag? Although this doesn't happen often, we've had to reach out to some contributors in the past.

Done

File: Documentation/operations/system_requirements.rst [hunk 3]
```diff
@@ -129,6 +129,15 @@ to ``/etc/systemd/network/01-no-dhcp.network`` and then
...
         systemctl restart systemd-networkd
 
+Ubuntu 22.04 on Raspberry Pi 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before running Cilium on Ubuntu 22.04 on a Raspberry Pi, please make sure to install the following package:
+
+.. code-block:: shell-session
+
+        sudo apt install linux-modules-extra-$(uname -r)
```
Comment (gentoo-root) [inline Documentation/operations/system_requirements.rst]:
```suggestion
        sudo apt install linux-modules-extra-raspi
```

As far as I understand (correct me if I'm wrong), linux-modules-extra-raspi pulls in the latest version of linux-modules-extra-*, updating it automatically with kernel updates. If you just install `linux-modules-extra-$(uname -r)`, Cilium will break again after a kernel update.

(Also I checked Ubuntu for amd64, and it doesn't require linux-modules-extra for xfrm_user and xfrm_interfaces, which were [apparently missing here](https://github.com/cilium/cilium/issues/20735#issuecomment-1201744660), so the "Ubuntu 22.04 on Raspberry Pi" header looks good to me.)

**Review by qmonnet — APPROVED** [review]
Thanks! Looks good to me assuming Maxim's comment gets addressed

**Review by gentoo-root — APPROVED** [review]
Thanks!

**Review by YutaroHayakawa — APPROVED** [review]

**Comment by fmunteanu** [comment]
@darox your documentation covers only Ubuntu, what about default RaspiOS?

**Comment by darox** [comment]
> @darox your documentation covers only Ubuntu, what about default RaspiOS?

I expect the default OS to require the same package. However, I don't have one at hand to test it.

**Comment by fmunteanu** [comment]
@darox it does not. This is a major roadblock since the kernel developers will not modify it, just to accommodate a third party app, as they describe Cilium. And they are correct. Right now RaspiOS users need to recompile the kernel, in order to accommodate the Cilium requirements, which is unrealistic for updates. I opened https://github.com/cilium/cilium/issues/23640 with Cilium.

**Comment by darox** [comment]
I didnt't know that. Thank you for the clarification.

**Comment by fmunteanu** [comment]
@darox not sure if you could help bring this to the attention of Cilium devs, it should be an easy fix.

**Comment by joestringer** [comment]
@fmunteanu Is there some change that you could propose as a PR? It's not clear to me how Cilium interacts with the kernel flag `CONFIG_ARM64_VA_BITS_39`.

**Comment by fmunteanu** [comment]
@joestringer is it possible to continue this discussion into https://github.com/cilium/cilium/issues/23640?

**Comment by joestringer** [comment]
@fmunteanu ah yes, that would be better than discussing on this PR. :+1:

**Pull request merged**